astropy/io/fits/convenience.py

```diff
@@ -489,7 +514,9 @@ def table_to_hdu(table):
 
             col.null = fill_value.astype(table[col.name].dtype)
     else:
-        table_hdu = BinTableHDU.from_columns(np.array(table.filled()))
+        table_hdu = BinTableHDU.from_columns(np.array(table.filled()), header=hdr)
+
+    coord_meta = table.meta.pop('__coordinate_columns__', None)
```

Comment:
Although it is somewhat different in that time is *not* a mixin column in a fits file.